test/e2e-tests/tests/pipelines_test.go

```diff
@@ -81,98 +51,219 @@ func init() {
 	ctx := CreateContext()
 
-	options, err := support.GetOptions()
+	config, err := support.GetConfig()
 	if err != nil {
-		t.Fatal(err.Error())
+		panic(err.Error())
+	}
+
+	if !config.GitOpsConfig.Enabled {
+		t.Skipf("skipping %v, Gitops pipeline not configured", t.Name())
+	}
+
+	if !config.GitFetchConfig.Enabled && !config.S3FetchConfig.Enabled {
+		t.Skipf("skipping %v, neither git_fetch or s3_fetch was not enabled", t.Name())
 	}
 
-	clients, err := support.CreateClients(options.ClusterNamespace)
+	clients, err := support.CreateClients(config.Namespace)
 	if err != nil {
 		t.Fatal(err.Error())
 	}
 
-	pipelineRunPaths := []string{GitOpsUpdateTensorflowHousingPipelineRunRelativePath, GitOpsUpdateBikeRentalsPipelineRunRelativePath}
-
-	gitURL, err := support.ParseGitURL(options.GitRepo)
+	gitURL, err := support.ParseGitURL(config.GitOpsConfig.Repo)
 	if err != nil {
 		t.Fatal(err.Error())
 	}
 
-	for _, path := range pipelineRunPaths {
-		pipelineRun, err := support.ReadFileAsPipelineRun(path)
+	pipelineRunFilesAndLabels := []struct {
+		pipelineRun string
+		label       string
+	}{
+		{pipelineRun: GitOpsUpdateTensorflowHousingPipelineRunRelativePath, label: "model-name=tensorflow-housing"},
+		{pipelineRun: GitOpsUpdateBikeRentalsPipelineRunRelativePath, label: "model-name=bike-rentals-auto-ml"},
+	}
+
+	for _, p := range pipelineRunFilesAndLabels {
+		pipelineRun, err := support.ReadFileAsPipelineRun(p.pipelineRun)
 		if err != nil {
 			t.Fatal(err.Error())
 		}
 
 		support.SetPipelineRunParam("gitServer", support.NewStringParamValue(gitURL.Server), &pipelineRun)
-		support.SetPipelineRunParam("gitApiServer", support.NewStringParamValue(options.GitAPIServer), &pipelineRun)
+		support.SetPipelineRunParam("gitApiServer", support.NewStringParamValue(config.GitOpsConfig.ApiServer), &pipelineRun)
 		support.SetPipelineRunParam("gitOrgName", support.NewStringParamValue(gitURL.OrgName), &pipelineRun)
 		support.SetPipelineRunParam("gitRepoName", support.NewStringParamValue(gitURL.RepoName), &pipelineRun)
-		support.SetPipelineRunParam("gitRepoBranchBase", support.NewStringParamValue(options.GitBranch), &pipelineRun)
+		support.SetPipelineRunParam("gitRepoBranchBase", support.NewStringParamValue(config.GitOpsConfig.Branch), &pipelineRun)
+
+		// we need to get the results from the pipeline run created by the ml ops pipeline, we also need
+		// to get the correct one that is for this model. Using the correct label and ensuring it was
+		// created after the tests have started then we can ensure we are getting the correct one
+		completedPipelineRuns, err := config.Clients.PipelineRun.List(ctx, metav1.ListOptions{
+			LabelSelector: p.label,
+		})
+		if err != nil {
+			t.Fatal(err.Error())
+		}
+
+		var targetPipelineRun *v1.PipelineRun
+		for i := 0; i < len(completedPipelineRuns.Items); i += 1 {
+			if completedPipelineRuns.Items[i].CreationTimestamp.After(TestStartTime) {
+				targetPipelineRun = &completedPipelineRuns.Items[i]
+			}
+		}
```

Comment:
We could also just "remember" the pipelineRunIDs, one for S3 and another one for Git. But I think this is OK too.

Reply:
Did this now, I also separated the gitops tests into two now aswell